subxt/src/lib.rs

```diff
@@ -33,23 +33,31 @@
 )]
 #![allow(clippy::type_complexity)]
 
+#[cfg(any(
+    all(feature = "web", feature = "native"),
+    not(any(feature = "web", feature = "native"))
+))]
+compile_error!("subxt: exactly one of the 'web' and 'native' features should be used.");
```

Comment:
This is a smart way to handle mutually exclusive features! Haven't seen that yet.

Reply:
the smart way is not to have mutually exclusive features :)

Reply:
Doesn't really have to be addressed in this PR but would be great if we could have APIs similar [to](https://github.com/paritytech/jsonrpsee/blob/master/client/ws-client/src/lib.rs#L127-#L131) to avoid having to deal with "mutually exclusive features`

but ofc it won't work for if something doesn't work for WASM.

Reply:
> the smart way is not to have mutually exclusive features :)

Yeah, ideally the features would be purely additive in general!

Probably I need to keep thinking about feature flags here, and also which flags are enabled by default (it's annoying re doc examples etc if you don't have feature flags enabled that you want to use but might be surmountable).

When I did this I figured that a non additive "native or web" feature and then everything else being additive (in terms of APIs exposed at least) felt like the nicest way to handle it (otherwise you end up with the non-additiveness being pushed to other features anyway like jsonrpsee-web and jsonrpsee-native, or lightclient-web and lightclient-native not playing well together).

Def open to more thoughts on all of this though!